presto-common/src/main/java/com/facebook/presto/common/predicate/vector/DoubleRangeVector.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.common.predicate.vector;
+
+import com.facebook.presto.common.predicate.TupleDomainFilter;
+import jdk.incubator.vector.DoubleVector;
+import jdk.incubator.vector.VectorMask;
+
+import static jdk.incubator.vector.VectorOperators.GE;
+import static jdk.incubator.vector.VectorOperators.GT;
+import static jdk.incubator.vector.VectorOperators.LE;
+import static jdk.incubator.vector.VectorOperators.LT;
+import static jdk.incubator.vector.VectorOperators.NE;
+
+public class DoubleRangeVector
+        extends TupleDomainFilter.DoubleRange
```

Comment:
what do we get out of extending DoubleRange? Seems like we don't use any of the DoubleRange functionality.  We should add those fields directly to DoubleRangeVector instead and make it independent of DoubleRange.

Reply:
DoubleRangeVector is an extention of the scalar equivalent and in the vector processing logic for handling the scalar tail part of execution we need the DoubleRange functionality to filter in scalar way.